src/frontend/sharedComponents/GPSPill.tsx

```diff
@@ -19,70 +28,104 @@ type GPSPillProps = {
     }
 );
 
-export const GPSPill = (props: GPSPillProps) => {
-  let textValue: string | React.ReactNode;
-  let IconToRender: React.FC;
-  // Replace spinner with static text in E2E mode to prevent rendering delays in tests
-  const isE2E = process.env.EXPO_PUBLIC_E2E_TEST === 'true';
+export const GPSPill = (props: Props) => {
+  let backgroundColor: string;
+  let icon: ReactNode;
+  let text: string;
 
   switch (props.status) {
-    case 'error':
-      textValue = '--';
-      IconToRender = GpsErrorIcon;
+    case 'error': {
+      backgroundColor = WARNING_RED;
+      text = '--';
+      icon = (
+        <View
+          testID={props.iconTestID}
+          style={{
+            backgroundColor: DARK_MAGENTA,
+            borderRadius: 50,
+            justifyContent: 'center',
+            alignItems: 'center',
+            width: 14,
+            height: 14,
+          }}>
+          <MaterialCommunityIcon name="exclamation" color={WHITE} size={12} />
+        </View>
+      );
+
       break;
-    case 'searching':
-      textValue = isE2E ? (
-        '...'
-      ) : (
-        <UIActivityIndicator size={20} color={WHITE} />
+    }
+
+    case 'searching': {
+      const isE2E = process.env.EXPO_PUBLIC_E2E_TEST === 'true';
+
+      backgroundColor = DARK_GREY;
+      text = '--';
+      icon = (
+        <View testID={props.iconTestID}>
+          <UIActivityIndicator
+            // Animations seem to cause issues with test performance on Browserstack
+            animating={!isE2E}
+            hidesWhenStopped={false}
```

Comment:
Testing to see if this approach can be used to avoid the animation perf overhead that animations have in browserstack, without needing to more drastically conditionally render something like what was being done prior.